src/prefect/task_engine.py

```diff
@@ -651,6 +662,7 @@ def transaction_context(self) -> Generator[Transaction, None, None]:
             store=ResultFactoryStore(result_factory=result_factory),
             overwrite=overwrite,
             logger=self.logger,
+            on_rollback_hooks=[__prefect_task_rollback_handler],
```

Comment:
I like that this approach keeps this "task-specific" behavior scoped to the task engine's transaction. 

in contrast to blurring the two more like making the transactions interface task-aware.

Reply:
Totally agreed, this was a clever design choice!